lib/rucio/daemons/conveyor/common.py

```diff
@@ -157,6 +165,254 @@ def run_conveyor_daemon(once, graceful_stop, executable, logger_prefix, partitio
                     activity_next_exe_time[activity] = time.time() + 1
 
 
+@transactional_session
+def next_transfers_to_submit(total_workers=0, worker_number=0, partition_hash_var=None, limit=None, activity=None, older_than=None, rses=None, schemes=None,
```

Comment:
One guiding principle (Although we broke it several times) was always to keep database-connected methods (e.g. everything with a session annotator) in the core. Is it needed to have this in conveyor/common? (Circular dependency?) I am not saying this is wrong or it should be changed, mostly wondering if we should try to "hold" this in core?

Reply:
No, let's leave it like this then. I don't find it easier to split these core files. Do I have to check core/transfer.py or core/transfer_execution.py now? 😄 Then I might as well just check conveyor/common.py. 👍

Reply:
I didn't know about this desire to put database-connected methods in core. I'll keep it in my head and, if I find an opportunity, I'll try to get rid of such accesses from outside the core in the future.

Reply:
It's not a rule written in stone, but that was always a bit the "idea" behind it. (Maybe we should write this down somewhere 😆 )